connector/src/main/java/com/google/cloud/hive/bigquery/connector/output/direct/DirectWriterContext.java

```diff
@@ -0,0 +1,144 @@
+    OVERWRITE,
+    ALL_ELSE
+  }
+
+  private final WritingMode writingMode = WritingMode.ALL_ELSE;
+
+  public DirectWriterContext(
+      BigQueryClient bigQueryClient,
+      BigQueryClientFactory bigQueryWriteClientFactory,
+      TableId destinationTableId,
+      Schema schema)
+      throws IllegalArgumentException {
+    this.bigQueryClient = bigQueryClient;
+    this.destinationTableId = destinationTableId;
+    this.tableToWrite = getOrCreateTable(destinationTableId, schema);
+    this.tablePathForBigQueryStorage =
+        bigQueryClient.createTablePathForBigQueryStorage(tableToWrite.getTableId());
+
+    this.writeClient = bigQueryWriteClientFactory.getBigQueryWriteClient();
+  }
+
+  /**
+   * This function determines whether the destination table exists: if it doesn't, we will create a
+   * table and Hive will directly write to it.
+   *
+   * @param destinationTableId the TableId, as was supplied by the user
+   * @param bigQuerySchema the bigQuery schema
+   * @return The TableId to which Hive will do the writing: whether that is the destinationTableID
+   *     or the temporaryTableId.
+   */
+  private BigQueryTable getOrCreateTable(
+      TableId destinationTableId, com.google.cloud.bigquery.Schema bigQuerySchema)
+      throws IllegalArgumentException {
+    if (bigQueryClient.tableExists(destinationTableId)) {
+      TableInfo destinationTable = bigQueryClient.getTable(destinationTableId);
+      com.google.cloud.bigquery.Schema tableSchema = destinationTable.getDefinition().getSchema();
+      Preconditions.checkArgument(
+          BigQueryUtil.schemaEquals(tableSchema, bigQuerySchema, /* regardFieldOrder */ false),
+          new BigQueryConnectorException.InvalidSchemaException(
+              "Destination table's schema is not compatible with query's" + " schema"));
+      return new BigQueryTable(destinationTable.getTableId(), false);
+    } else {
+      return new BigQueryTable(
+          bigQueryClient.createTable(destinationTableId, bigQuerySchema).getTableId(), true);
+    }
+  }
+
+  public void commit(List<String> streamNames) {
+    if (writingMode.equals(WritingMode.IGNORE_INPUTS)) return;
+    BatchCommitWriteStreamsRequest.Builder batchCommitWriteStreamsRequest =
+        BatchCommitWriteStreamsRequest.newBuilder().setParent(tablePathForBigQueryStorage);
+    for (String streamName : streamNames) {
+      batchCommitWriteStreamsRequest.addWriteStreams(streamName);
+    }
+    BatchCommitWriteStreamsResponse batchCommitWriteStreamsResponse =
+        writeClient.batchCommitWriteStreams(batchCommitWriteStreamsRequest.build());
+
+    if (!batchCommitWriteStreamsResponse.hasCommitTime()) {
+      throw new BigQueryConnectorException(
+          "BigQuery writer failed to batch commit its BigQuery write-streams");
+    }
+
+    LOG.info(
+        "BigQuery writer has committed at time: {}",
+        batchCommitWriteStreamsResponse.getCommitTime());
+
+    if (writingMode.equals(WritingMode.OVERWRITE)) {
```

Comment:
Where do we set the writingMode? From what I see, we always default to `WritingMode.ALL_ELSE`?

Reply:
Good point. I've removed the redundant code.